internals/schema_migration.py

```diff
@@ -701,3 +701,63 @@ def get_template_data(self, **kwargs) -> str:
         count += len(stages_to_update)
 
     return f'{count} subject line fields migrated.'
+
+
+class MigrateLGTMFields(FlaskHandler):
+
+  def get_template_data(self, **kwargs) -> str:
+    """Migrates old Feature subject lgtms to their respective votes."""
+    self.require_cron_header()
+
+    count = 0
+    votes_to_create = []
+    for f in Feature.query():
+      if (f.i2e_lgtms and not self.has_existing_vote(f.i2e_lgtms, GATE_API_ORIGIN_TRIAL, f)):
+        # i2e_lgtms (Intent to Experiment)'s gate_type is GATE_API_ORIGIN_TRIAL.
+        votes_to_create.append(self.create_new_vote(
+            f.i2e_lgtms, GATE_API_ORIGIN_TRIAL, f))
+        count += 1
+
+      if (f.i2s_lgtms and not self.has_existing_vote(f.i2s_lgtms, GATE_API_SHIP, f)):
+        # i2s_lgtms (Intent to Ship)'s gate_type is GATE_API_SHIP.
+        votes_to_create.append(self.create_new_vote(
+            f.i2s_lgtms, GATE_API_SHIP, f))
+        count += 1
+
+    ndb.put_multi(votes_to_create)
+    return f'{count} lgtms fields migrated.'
+
+  def has_existing_vote(self, email, gate_type, f):
```

Comment:
I think that calling this in a loop in prod will timeout.  We have over 2000 Feature entities on prod.  I think that making 2000 * 2 * 2 = 8000 queries could exceed the time limit for a request.
It would be better to prefetch all existing Vote entities and put them into a dictionary keyed by feature_id with each value being a list of Votes. 

I don't think that you need to fetch the Gates because Vote has feature_id and gate_type.

Reply:
Addressed